### ISSUE TITLE
BO - Logs / the name and first name variables are not interpreted in the logs email

### ISSUE THREAD

**Issue by SD1982** [PrestaShop/PrestaShop#22091]
#### Describe the bug

the name and first name variables are not interpreted in the logs email

#### Expected behavior

The varibles should be interpreted

#### Steps to Reproduce

PS => The SMTP parameters should be correctly set

1. Go to Advanced parameters / logs
2. Set the "minimum security level" to 1
3. Make some action to generate log ( try to import a localisation pack for exemple)
4. Go to your mail box and see error => The "firstname" and "lastname" variables are not interpreted

**Screenshots**
![image](https://user-images.githubusercontent.com/40635814/100211390-6117e100-2f0c-11eb-91ec-97586fa301f9.png)

#### Additional information

* PrestaShop version: 1.7.6.9 && 1.7.7.x && develop 
* PHP version: 7.1

**Comment by LouiseBonnard** [comment]
You're right, thanks @SD1982, variables should be interpreted in the email. Yet, this email is intended for the merchant/employee so it is unnecessary to add the first and last names in each email. A simple `Hi,` (instead of `Hi {firstname} {lastname},`) in `Emails.Body` looks enough and less formal. ;-)

Here are the other emails (modern template) this change could be applied:

- [ ] forward_msg
- [ ] log_alert
- [x] order_customer_comment (fixed by https://github.com/PrestaShop/mjml-theme-converter/pull/8)
- [ ] productoutofstock
- [ ] reply_msg ?
- [ ] test

**Comment by ghost** [comment]
Hi @LouiseBonnard i made this one : https://github.com/PrestaShop/mjml-theme-converter/pull/11

**Comment by PrestaEdit** [comment]
The Pull Request seems merged, I think we can close this issue.

**Comment by matks** [comment]
I think we might need to re-build the mail theme using https://github.com/PrestaShop/mjml-theme-converter/ ?

**Comment by florine2623** [comment]
In PS 8.1.x, the variables are not displayed :

<img width="567" alt="Screenshot 2023-12-27 at 15 39 37" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/d6656be8-27e1-40f6-841a-5c819a7d6efd">

**Comment by boo-code** [comment]
@florine2623's 2023 observation still holds on 9.2.x, and the reason the earlier fix did not land is that it was applied to one theme only.

`PrestaShopLogger::sendByMail()` sends with an empty variable list:

```php
Mail::Send(
    (int) Configuration::get('PS_LANG_DEFAULT'),
    'log_alert',
    ... ,
    [],
    $to
);
```

so nothing can substitute a placeholder. `mails/themes/modern/core/log_alert.html.twig` already reads `{{ 'Hi,'|trans({}, 'Emails.Body', locale) }}`, which is @LouiseBonnard's decision applied. `mails/themes/classic/core/log_alert.html.twig` still reads `{{ 'Hi {firstname} {lastname},'|trans({}, 'Emails.Body', locale) }}`. That is the whole remaining bug — one line in the classic theme.

The rest of the checklist in the first comment is already correct and should be left alone. Each of those senders does supply the variables, so their placeholders resolve:

| template | sender | supplies `{firstname}` |
|---|---|---|
| `log_alert` | `PrestaShopLogger::sendByMail()` | no — passes `[]` |
| `productoutofstock` | `StockManager`, per employee | yes |
| `forward_msg` | `ForwardCustomerThreadHandler` | yes |
| `reply_msg` | `ReplyToCustomerThreadHandler` | yes |
| `test` | `MailPreviewVariablesBuilder:126-127` | yes |
| `contact_form` | `contactform` module | yes |

Comparing the two themes across all core templates, the divergence actually runs both ways — `contact_form`, `productoutofstock` and `test` carry the placeholder in modern and not in classic — but since their senders provide the values, removing it there would delete working personalisation rather than fix anything.

One separate leftover worth its own decision: `contact_form` initialises `{firstname}` to `''` and only fills it when the sender is a logged-in customer, so a guest submission renders "Hi ,". Different defect from the literal placeholder, so I have not touched it.

Fix prepared on `9.2.x`, with a unit test asserting the `log_alert` template of every shipped theme uses no placeholder the sender cannot provide. Reverting the template alone fails the classic case and leaves modern passing.

I am at the open-PR limit, so I cannot open it yet. I will link it here as soon as a slot frees.
